src/Source/VectorTilesSource.js

```diff
@@ -126,6 +126,13 @@ class VectorTilesSource extends TMSSource {
             }
         });
     }
+
+    onLayerAdded(options) {
+        super.onLayerAdded(options);
+        const keys = Object.keys(this.styles);
+
+        keys.forEach((k) => { this.styles[k].parent = options.out.style; });
+    }
```

Comment:
What happens if a source is used by several layers?

Reply:
Effectively, there's a issue. I'll look into it. But there's a problem with parent style property.

Reply:
I created the issue, the Style architecture needs refactoring.